addons/mail/static/src/model/model_manager.js

```diff
@@ -1075,6 +1081,14 @@ class ModelManager {
             if (!field) {
                 throw new Error(`Cannot create/update record with data unrelated to a field. (model: "${Model.modelName}", non-field attempted update: "${fieldName}")`);
             }
+
+            if ((!options.allowWriteReadonly) && field.readonly) {
```

Comment:
no need for extra parenthesis

Reply:
I would also swap the order of those condition, the first one `field.readonly` is more important to understand the context of the condition. `!options.allowWriteReadonly` is just an extra check.